js/tidex2.js

```diff
@@ -0,0 +1,960 @@
+        };
+        const response = await this.publicGetTicker (this.extend (request, params));
+        const result = this.safeValue (response, 'result');
+        //
+        //     {
+        //         "code":200,
+        //         "success":true,
+        //         "message":"",
+        //         "result":{
+        //             "name":"YFI_USDT",
+        //             "bid":"21607.842223",
+        //             "ask":"21635.50514299",
+        //             "open":"21530.31",
+        //             "high":"22619.38",
+        //             "low":"21193.59",
+        //             "last":"21644.23",
+        //             "volume":"188.4555124",
+        //             "deal":"4075612.142786252",
+        //             "change":"1"
+        //         }
+        //     }
+        //
+        return this.parseTicker (result, market);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = 1, max = 100
+        }
+        const response = await this.publicGetDepthResult (this.extend (request, params));
+        //
+        //     {
+        //         "asks":[
+        //             ["22064.13194312","0.25742"],
+        //         ],
+        //         "bids":[
+        //             ["22017.12205596","0.11"],
+        //         ]
+        //     }
+        //
+        return this.parseOrderBook (response, symbol);
+    }
+
+    parseTrade (trade, market = undefined) {
+        // fetchTrades
+        //     {
+        //         "tid":135762344,
+        //         "date":1646294384,
+        //         "price":"21991.91",
+        //         "type":"buy",
+        //         "amount":"0.0024",
+        //         "total":"52.780584"
+        //     }
+        //
+        const timestamp = this.safeTimestamp (trade, 'date');
+        const side = this.safeString (trade, 'type');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString (trade, 'amount');
+        const id = this.safeString (trade, 'tid');
+        const symbol = this.safeSymbol (undefined, market);
+        return this.safeTrade ({
+            'id': id,
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'order': undefined,
+            'type': undefined,
+            'takerOrMaker': undefined,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (since === undefined) {
```

Comment:
we should remove lines 487 to 488 because it seems that the *since* parameter taken by the exchange in the request and is misleadingly named - it refers to the *tid* property of each trade which in this case is the ever-increasing exchange-specific identifier for each trade and not what we consider the unified term `since` to be in CCXT (we take `since` to be an integer UTC timestamp in milliseconds and return all trades/orders after that point in time) 

so when you pass *since* to the request it returns all trades with a *tid* identifier greater than the one you supplied